plugins/kubernetes/app/models/kubernetes/client_config_file.rb

```diff
@@ -85,6 +90,34 @@ def parse_contexts
       end
     end
 
+    def ext_file_path(path)
+      File.join(File.dirname(@filepath), path)
+    end
+
+    def fetch_cluster_ca_data(cluster)
+      if cluster.key?('certificate-authority')
```

Comment:
not a big fan of repeating keys ... how about a if assignment or assign + check ?

```Ruby
if data = cluster['certificate-authority']
  ... parse data ...
end
```